### PR TITLE
decouple server from namer, reuse for a tpdb fake

### PR DESCRIPTION
**Pull request by 4c0d3r**
Decouple webserver from namer for easy reuse.
Reuse webserver to make a ParrotWebserver.
Use ParrotWebserver for a fake tpdb for testing.

Todo: 
finish configuring response from fake tpdb
update the test to use Fake_tpdb.

### CODE REVIEW

File: test/web/namer_web_pageobjects.py [hunk 1]
```diff
@@ -164,7 +164,7 @@ def file_size(self) -> Assertion['FailedItem']:
 
 
 class FailedPage():
-    __driver = WebDriver
```
Comment (4c0d3r) [inline test/web/namer_web_pageobjects.py:167]: stupid bug

File: namer/web/server.py [hunk 2]
```diff
@@ -32,21 +32,22 @@ class WebServer:
         'font/woff2': '.woff2',
     }
 
-    def __init__(self, host: str, port: int, webroot: Optional[str], blueprints: List[Blueprint]):
+    def __init__(self, host: str, port: int, webroot: Optional[str], blueprints: List[Blueprint], static_url: Optional[str] = 'public'):
         self.__host = host
         self.__port = port
         self.__webroot = webroot
         self.__path = '/' if not self.__webroot else self.__webroot
-        self.__app = Flask(__name__, static_url_path=self.__path, static_folder='public', template_folder='templates')
```
Comment (4c0d3r) [inline namer/web/server.py:40]: needed to let a the "catch all" route to work in ParrotWebserver

File: test/web/namer_web_test.py [hunk 3]
```diff
@@ -15,10 +16,7 @@
 from test.namer_watchdog_test import make_locations, new_ea, prepare
 from test.web.namer_web_pageobjects import FailedPage
 from namer.types import NamerConfig, default_config
-
-_browser = Chrome
-_options = ChromeOptions
-_manager = ChromeDriverManager
```
Comment (4c0d3r) [inline test/web/namer_web_test.py:21]: Removed for now as different browsers will require a different function to configure the browser, let me know if we need this on you desktop now Dirty

Comment (DirtyRacer1337) [reply test/web/namer_web_test.py:21]: Don't like this global vars too, maybe do it like that

I don't have chrome, but edge is default for windows, so I'm used it

```python
from selenium.webdriver import Chrome as Browser, ChromeOptions as BrowserOptions
from selenium.webdriver.chrome.service import Service
from webdriver_manager.chrome import ChromeDriverManager as BrowserDriverManager
```

File: namer/types.py [hunk 4]
```diff
@@ -16,8 +16,10 @@
 from typing import List, Optional, Sequence
 
 from loguru import logger
-from pathvalidate import Platform, sanitize_filename
-from requests_cache import BACKEND_CLASSES, BaseCache, CachedSession
```
Comment (DirtyRacer1337) [inline namer/types.py:20]: old is better

Comment (4c0d3r) [reply namer/types.py:20]: yeah, it's an error in pylance though, not sure how to ignore the error

File: test/web/namer_web_test.py (hunk 3 above)
Comment (4c0d3r) [reply test/web/namer_web_test.py:21]: I plan to use a single different functions for each supported browser  -- as each browser has it's own setup requirements.

File: namer/types.py (hunk 4 above)
Comment (DirtyRacer1337) [reply namer/types.py:20]: looks like pylance lying to you, `pathvalidate` have this imports in `__init__`, https://github.com/thombashi/pathvalidate/blob/master/pathvalidate/__init__.py

File: namer/web/server.py [hunk 5]
```diff
@@ -32,21 +32,22 @@ class WebServer:
...
+        self.__app = Flask(__name__, static_url_path=self.__path, static_folder=static_url, template_folder='templates')
         self.__blueprints = blueprints
 
         self.__add_mime_types()
         self.__register_blueprints()
         self.__make_server()
+        print(self.__app.url_map)
```
Comment (DirtyRacer1337) [inline namer/web/server.py:46]: better have method like `get_url_map()`, or it's just temp for debug?

File: test/web/namer_web_test.py (hunk 3 above)
Comment (4c0d3r) [reply test/web/namer_web_test.py:21]: fixed.

File: namer/web/server.py (hunk 5 above)
Comment (4c0d3r) [reply namer/web/server.py]: removed

**Pull request merged**